app/operations/story_operations.rb

```diff
@@ -56,34 +56,40 @@ class Destroy < BaseOperations::Destroy; end
   class DestroyAll < BaseOperations::DestroyAll; end
 
   class ReadAll
+    delegate :past_iterations, :current_iteration_start, to: :iterations
+
     def self.call(*args)
       new(*args).run
     end
 
     def initialize(project:)
-      @story_scope = project.stories.with_dependencies
       @project = project
-      @iterations = Iterations::ProjectIterations.new(project: project)
     end
 
     def run
       {
         active_stories: active_stories,
-        past_iterations: @iterations.past_iterations
+        past_iterations: past_iterations
       }
     end
 
     private
 
+    attr_reader :project
+
     def active_stories
-      order(@story_scope.where("state != 'accepted' OR
-        accepted_at >= ?", @iterations.current_iteration_start))
+      @active_stories ||= begin
+        project
+          .stories
+          .with_dependencies
+          .where("state != 'accepted' OR accepted_at >= ?", current_iteration_start)
+          .order('updated_at DESC')
+          .limit(ENV['STORIES_CEILING'].presence&.to_i)
```

Comment:
What do you think about extracting the `ENV['STORIES_CEILING'].presence&.to_i` to a method?

Reply:
I want to understand why this ceiling is there? 

Active stories need to be limited?

Reply:
I'm removing this. We are lazy loading done stories nowadays.

Reply:
As spoke offline, we won't need this `limit` anymore.